src/DescriptorsCommand.cc

```diff
@@ -668,6 +667,29 @@ int FindDescriptor::construct_protobuf(
     return 0;
 }
 
+void FindDescriptor::populate_blobs(const std::string& set_path,
+                                    const Json::Value& results,
+                                    Json::Value& entities,
+                                    protobufs::queryMessage &query_res)
+{
+    if (get_value<bool>(results, "blob", false)) {
+
+        VCL::DescriptorSet* set = _dm->get_descriptors_handler(set_path);
+        int dim = set->get_dimensions();
+
+        for (auto& ent : entities) {
```

Comment:
Isn't this separation making you traverse the entity list twice? Once for properties and once for blobs. WOuld it work if you put the loop around entities outside the convert_properties and populate_blobs functions?

Reply:
no, the separation keeps the logic as is. The list is traversed twice in any case because the convert_properties functions aim to trim some of the results if needed as a last step. I am not sure we want to couple the logic for insertion the blobs to the logic to prune/convert the name of some properties. I couldn't think in a way of doing this without looking horribly coupled.